src/main/webapp/app/overview/exercise-details/difficulty-badge.component.ts

```diff
@@ -0,0 +1,85 @@
+import { Component, HostBinding, Input, OnInit, OnDestroy } from '@angular/core';
+import { DifficultyLevel, Exercise } from 'app/entities/exercise';
+import { TranslateService, LangChangeEvent } from '@ngx-translate/core';
+import { Subscription } from 'rxjs/Subscription';
+
+@Component({
+    selector: 'jhi-difficulty-badge',
+    templateUrl: './difficulty-badge.component.html'
+})
+export class DifficultyBadgeComponent implements OnInit, OnDestroy {
+    @Input() exercise: Exercise;
+    @Input() showNoLevel: true;
+    public translatedDifficulty: string;
+    public badgeClass: string;
+    private translateSubscription: Subscription;
+
+    @HostBinding('class.badge-easy')
+    public get isEasy(): boolean {
+        return this.exercise.difficulty === DifficultyLevel.EASY;
+    }
+
+    @HostBinding('class.badge-medium')
+    public get isMedium(): boolean {
+        return this.exercise.difficulty === DifficultyLevel.MEDIUM;
+    }
+
+    @HostBinding('class.badge-hard')
+    public get isHard(): boolean {
+        return this.exercise.difficulty === DifficultyLevel.HARD;
+    }
+
+    constructor(private translateService: TranslateService) {
+
+    }
+
+    ngOnInit(): void {
+        this.translateSubscription = this.translateService.onLangChange.subscribe((event: LangChangeEvent) => {
+            this.translateBadge();
+        });
+        this.translateBadge();
+        this.setBadgeColor();
+    }
+
+    ngOnDestroy(): void {
+        this.translateSubscription.unsubscribe();
+    }
+
+    private setBadgeColor() {
+        switch (this.exercise.difficulty) {
+            case DifficultyLevel.EASY:
+                this.badgeClass = 'badge-success';
+                break;
+            case DifficultyLevel.MEDIUM:
+                this.badgeClass = 'badge-warning';
+                break;
+            case DifficultyLevel.HARD:
+                this.badgeClass = 'badge-danger';
+                break;
+            default:
+                if (this.showNoLevel) {
+                    this.badgeClass = 'badge-info';
+                    this.translatedDifficulty = this.translateService.instant('arTeMiSApp.exercise.noLevel');
+                }
+        }
+    }
+
+    private translateBadge() {
```

Comment:
Personally, I'd prefer use an object instead of a switch, because code would be much shorter.

So you create an object with all the properties:

```
readonly const levelObject = { 
  [DifficultyLevel.EASY]: {translatedDifficulty: t('easy'), badgeClass: 'badge-success'},
  [DifficultyLevel.MEDIUM]: {translatedDifficulty: t('medium'), badgeClass: 'badge-warning'},
  [DifficultyLevel.HARD]: {translatedDifficulty: t('hard'), badgeClass: 'badge-danger'},
  default: {translatedDifficulty: t('noLevel'), badgeClass: 'badge-info'}
}
```

and your functions would be:

```
private translateBadge() {
   return levelObject[this.exercise.difficulty] || this.showNoLevel ? levelObject['default'] : null
}
```

(but I think you can remove the `this.showNoLevel` check

And I would also curry the translation function, so, as you saw, I put t, and you would define t as

```
const t => (level: str) => this.translateService.instant(`arTeMiSApp.exercise.${level}`);
```

Up to you, just a suggestion on a different approach :)

Reply:
i removed the second switch case. I would leave it like that for the moment